src/core/plugins

```diff
@@ -0,0 +1,306 @@
+    return 0
+  fi
+}
+
+# Removes dups from a csv string
+csv_remove_dups() {
+  local csv="${1}"
+  local -a array
+
+  array=$( echo "${csv}" | tr ',' '\n' | sort -u | tr '\n' ' ' )
+  array=( ${array% } )
+  echo ${array[@]}
+}
+
+
+## Helper functions for plugin cli actions
+
+# Obtain the list of plugins in a directory
+plugins_list() {
+  local plugin_dir="$1"
+  local p_names=( $( ls -1 "${plugin_dir}" | tr '\n' ' ') )
+  echo ${p_names[@]}
+}
+
+# Validate action (list, enable, disable)
+plugin_validate_action() {
+  case "${1}" in
+    "list"|"enable"|"disable")
+      return 0
+    ;;
+    *)
+      return 1
+    ;;
+  esac
+}
+
+# Obtain a list of plugin by status
+get_plugin_status() {
+  local list_status="${1}"
+  local -a plugins
+
+  case "${list_status}" in
+    "available")
+      plugins+=( $( plugins_list "${PLUGIN_A_DIR}" ) )
+    ;;
+    "enabled")
+      plugins+=( $( plugins_list "${PLUGIN_E_DIR}" ) )
+    ;;
+    "disabled")
+      plugins_a=( $( plugins_list "${PLUGIN_A_DIR}" ) )
+      plugins_e=( $( plugins_list "${PLUGIN_E_DIR}" ) )
+      # Loop through the available plugins
+      for p_a in ${plugins_a[@]}; do
+        # Copy the list of enabled plugins removing the available in turn
+        c_p_e=( ${plugins_e[@]//${p_a}} )
+        # If both arrays are the same then the looped plugin is disabled
+        if [[ ${#plugins_e[@]} == ${#c_p_e[@]} ]]; then
+            plugins+=( ${p_a} )
+        fi
+      done
+    ;;
+  esac
+
+  echo "${plugins[@]}"
+}
+
+# Prints the status of plugins
+print_plugins() {
+  local list_status="${1}"
+  local plugins
+
+  case "${list_status}" in
+    "available"|"enabled"|"disabled")
+      plugins=$( get_plugin_status "${list_status}" )
+    ;;
+    *)
+      echo -e "Error: Invalid list status argument: ${list_status}\n" >&2
+      print_usage
+      return 1
+    ;;
+  esac
+
+  if [[ ${#plugins} == 0 ]]; then
+    log ERROR "No ${list_status} plugins found."
+  else
+    plugins=$( echo "${plugins[@]}" | tr " " "," )
+    echo "Plugins ${list_status}: ${plugins%,}"
+    log INFO "Plugins ${list_status}: ${plugins%,}"
+  fi
+  return 0
+}
+
+# Validate plugin names. "all" is valid (represent all available)
+validate_plugin() {
+  local plugins="$1"
+  local -a plugins_a
+
+  # Validate "all", it is not valid to combine "all" with other plugins
+  if [[ "${plugins}" == "all" ]]; then
+    return 0
+  fi
+  
+  # Getting the list of available plugins
+  plugins_a=( $( get_plugin_status "available" ) )
+
+  # Create array with the provided plugins, remove dups
+  local -a p_v=( $( csv_remove_dups "${plugins}" ) )
+  
+  # Validate provided plugin with available
+  for p in ${p_v[@]}; do
+    if ! in_array ${p} ${plugins_a[@]}; then
+        return 1
+    fi
+  done
+
+  return 0
+}
+
+# Enable Plugin support in recap.conf
+enable_plugin_conf() {
+   local recap_conf="/etc/recap.conf"
+
+   # Only perform changes when not already defined
+   if [[ "${USEPLUGINS}" == "no" ]]; then
```

Comment:
To reduce nesting, test for USEPLUGINS=yes, and return.  Then the rest of this function can be 1-less-nest.